.husky/pre-commit

```diff
@@ -1,4 +1,6 @@
 npm run format
 npm run lint
 npm run leaks
-npm test
\ No newline at end of file
+npm test
+(cd examples/es6 && npm i && npm run build)
+(cd examples/commonjs && npm i)
```

Comment:
I must say I find it very annoying to run all those things before every commit

Reply:
its pretty fast tho, I can include that in the build but if we have package-lock updates or example build failure you wouldn't know until it will fail in the build. Easier if precommit will enforce it no?

Reply:
I think we should not enforce everyone to run it on every commit, 
If you're asking me, I prefer it to fail in the PR from time to time rather then running all those things on every commit